keps/sig-windows/5100-windows-dsr-and-overlay-support/README.md

```diff
@@ -0,0 +1,925 @@
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+Overlay networking mode is not compatible with dualstack networking on Windows.
+
+If kube-proxy is started with both overlay networking mode and dualstack networking enabled, a warning message will be added and ip address space with be downgraded to ipv4 only. This is existing behavior and has not caused and any reported issues.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+Enabling DSR and overlay networking mode support in Windows kube-proxy both have very little risk.
+
+For DSR, the Windows Host Network Service handles all of the logic for managing network traffic; kube-proxy only needs to specify if DSR should be used when creating/sycing load balancer rules.
+Additionally, DSR must be enabled with a kube-proxy command switch switch (--enable-dsr=true) disabling DSR is can be performed by redeploying kube-proxy on Windows nodes.
+
+Overlay networking support in Windows has been used in the Windows CI pipelines running release-informing jobs for many releases and is considered stable.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+Since the functionality is already implemented, the design details section will cover the current implementation.
+
+### DSR Enablement
+
+DSR is enabled by passing `--enable-dsr=true` as a command line switch to the Windows kube-proxy.
+Prior to GA, kube-proxy will ensure that `WinDSR=true` is specified in the feature-gates and will fail to start if DSR is enabled without that.
+
+Checks for ternminating and service enpoints handle DSR traffic differently than non-DSR traffic to adhere to behavior defined in [KEP-1669: Proxy Terminating Endpoints](https://github.com/kubernetes/enhancements/issues/1669)
+- Local endpoints will be skipped when determining if all endpoints for a service are terminated if DSR is enabled and service type is load balancer.
+- Non-local endpoints will be skipped when considering if all endpoints for a service are non-serving if DSR is enabled and service type is load balancer.
+
+Flags passed to HNS calls used for the following operators will be updated to include a flag indicating if DSR is enabled for all get, create, and update loadbalancer HNS calls.
```

Comment:
Maybe define what HNS is?

Reply:
I thought I did this but I must have removed that section in some edits. Will add